stingray/io.py

```diff
@@ -653,3 +660,66 @@ def get_btis(gtis, start_time=None, stop_time=None):
 def gti_len(gti):
     """Return the total good time from a list of GTIs."""
     return np.sum([g[1] - g[0] for g in gti])
+
+def _save_pickle_object(object, filename):
+    pickle.dump(object, open(filename, "wb" ))
+
+def _retrieve_pickle_object(filename):
+    return pickle.load(open(filename, "rb" ) )
+
+def _save_hdf5_object(object, filename):
+    pass
+
+def _retrieve_hdf5_object(object, filename):
+    pass
+
+def _save_ascii_object(object, filename):
+    pass
+
+def _retrieve_ascii_object(object, filename):
+    pass
+
+def save_object(object, filename, format):
+    """
+    Pickle a class instance.
+
+    Parameters
+    ----------
+    object: a class instance
```

Comment:
I will surprise you: `hdf5` is awesome, but it has some drawbacks (no unicode and longdouble support, for example). So, for default value, I would choose `pickle`! Only for internal use, of course. For final products, `fits` is still the way to go, I guess. And `ascii` is desirable as an option, bu,t for most observations, disk space would puff up immediately to hundreds of GB if we want to maintain some precision.